commands/moderation/timeout.js

```diff
@@ -0,0 +1,175 @@
+const Discord = require('discord.js');
+const dateformat = require('dateformat');
+const ms = require('ms');
+const {verifyReasonLength} = require('../../helpers/stringHelpers');
+
+module.exports = {
+  name: 'timeout',
+  description: 'Timeout a user',
+  guildOnly: true,
+  staffOnly: true,
+  minRole: 'Moderator',
+
+  async execute(msg, args, con) {
+    const {status, err, toTimeout, duration, reason} = canTimeout(msg, args);
+    if (!status) {
+      return msg.reply({content: err});
+    }
+
+    if (!verifyReasonLength(msg.content, msg)) return;
+
+    timeoutUser(msg, toTimeout, duration, reason);
+    auditLogTimeout(msg, toTimeout, duration, reason);
+    recordTimeoutInDB(msg, toTimeout, duration, reason, con);
+    try {
+      await dmTheUser(msg, toTimeout, duration, reason);
+    } catch (e) {
+      return msg.reply(`${e.name}: ${e.message}`);
+    }
+  },
+};
+
+// Validates data.
+function canTimeout(msg, args) {
+  const data = {
+    status: false,
+    err: null,
+    toTimeout: null,
+    duration: null,
+    reason: null,
+  };
+
+  // Check if targeted user is valid and can be timed out.
+  data.toTimeout =
+    msg.mentions.members.first() || msg.guild.members.cache.get(args[0]);
+  if (!data.toTimeout) {
```

Comment:
More for the sake of consistency and the snarky messages I think we could use the `hasTargetUser` helper function here.

Reply:
@lyallstewart After some testing, I've found that it wouldn't be a good idea to use the `hasTargetUser` helper. I've tried implementing it and the only way to do so without an error is to send **two** replies: one would be Fede's snarky message and the other would be another error message. This is because of the way that the input is parsed and destructured. The following line (line 15) must be executed if the command is used incorrectly:
```javascript
return msg.reply({content: err});
```
This is the same reason why I think we haven't incorporated the `hasTargetUser` helper function into the mute and unmute commands.